### PR TITLE
[pt1][quant] Add `quantize_linear_per_channel`

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#20765 [pt1][quant] Add `quantize_linear_per_channel`**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15435455/)

att

Differential Revision: [D15435455](https://our.internmc.facebook.com/intern/diff/D15435455/)

### CODE REVIEW

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -2546,6 +2546,11 @@
   dispatch:
     CPU: quantize_linear_cpu
 
+- func: quantize_linear_per_channel(Tensor self, Tensor scales, Tensor zero_points, Tensor axis, ScalarType dtype) -> Tensor
```
Comment (dzhulgakov) [inline aten/src/ATen/native/native_functions.yaml]: why  `axis` is a 1-d tensor and not `int[]` like other sizes?

**Review by dzhulgakov — CHANGES_REQUESTED** [review]

File: aten/src/ATen/native/native_functions.yaml (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/native/native_functions.yaml]: was trying to be consistent.. I will change to `int[]`

File: aten/src/ATen/native/quantized/QTensor.cpp [hunk 2]
```diff
@@ -7,16 +7,37 @@
...
 
+Tensor quantize_linear_per_channel_cpu(
+    const Tensor& self, const Tensor& scales, const Tensor& zero_points,
+    IntArrayRef axis, ScalarType dtype) {
+  TORCH_CHECK(scales.dim() == 1, "scale tensor must have dimension 1");
+  TORCH_CHECK(zero_points.dim() == 1,
+              "zero_points tensor must have dimension 1");
+  TORCH_CHECK(scales.numel() == zero_points.numel(),
+              "number of elements in scales and zero_points must match");
+  TORCH_CHECK(axis.size() == 1, "axis tensor must have size 1");
```
Comment (dzhulgakov) [inline aten/src/ATen/native/quantized/QTensor.cpp]: nit - it's not a tensor. And change the message that it's not yet supported (is it right?)

File: aten/src/ATen/quantized/Quantizer.cpp [hunk 3]
```diff
@@ -223,7 +223,7 @@ Tensor dequantize_tensor_per_channel_affine(Tensor qtensor,
                                             Tensor rtensor,
                                             std::vector<float> scales,
```
Comment (dzhulgakov) [inline aten/src/ATen/quantized/Quantizer.cpp]: btw, since noticed - should they all be `const vector&`?

**Review by dzhulgakov — APPROVED** [review]

File: aten/src/ATen/native/quantized/QTensor.cpp (hunk 2 above)
Comment (jerryzh168) [reply aten/src/ATen/native/quantized/QTensor.cpp]: we don't know whether we'll support size != 1 but size 1 is only thing that we'll support now.

**Pull request closed without merging**